libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestRequest.java

```diff
@@ -378,23 +473,152 @@ public static RestRequest getRequestForSearchResultLayout(String apiVersion, Lis
 		StringBuilder path = new StringBuilder(RestAction.SEARCH_RESULT_LAYOUT.getPath(apiVersion));
 		path.append("?q=");
 		path.append(URLEncoder.encode(toCsv(objectList).toString(), UTF_8));
-		return new RestRequest(RestMethod.GET, path.toString(), null);	
-	}	
-	
-	/**
-	 * Jsonize map and create a RequestBody out of it
-	 * @param fields
+		return new RestRequest(RestMethod.GET, path.toString());
+	}
+
+    /**
+	 * Composite request
+	 * See https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_composite.htm
+	 *
+     * @param apiVersion
+     * @param allOrNone
+	 * @param refIdToRequests    Sorted map of reference id to refIdToRequests
 	 * @return
-	 * @throws UnsupportedEncodingException
 	 */
-	private static RequestBody prepareFieldsData(Map<String, Object> fields)
-			throws UnsupportedEncodingException {
-		if (fields == null) {
-			return null;
-		}
-		else {
-			return RequestBody.create(MEDIA_TYPE_JSON, new JSONObject(fields).toString());
+	public static RestRequest getCompositeRequest(String apiVersion, boolean allOrNone, SortedMap<String, RestRequest> refIdToRequests) throws JSONException {
```

Comment:
Composite needs a list of RestRequests with a reference id for each one - so we are using a sorted map of reference id to RestRequest.